apps/extension/src/wasm-task.ts

```diff
@@ -0,0 +1,58 @@
+import {
+  Action,
+  TransactionPlan,
+  WitnessData,
+} from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/transaction/v1alpha1/transaction_pb';
+import { JsonValue } from '@bufbuild/protobuf';
+
+interface WasmTaskInput {
+  transactionPlan: JsonValue;
+  witness: JsonValue;
+  fullViewingKey: string;
+  actionId: number;
+}
+
+const workerListener = ({ data }: { data: WasmTaskInput }) => {
+  const {
+    transactionPlan: transactionPlanJson,
+    witness: witnessJson,
+    fullViewingKey,
+    actionId,
+  } = data;
+
+  // Deserialize payload
+  const transactionPlan = TransactionPlan.fromJson(transactionPlanJson);
+  const witness = WitnessData.fromJson(witnessJson);
+
+  void executeWorker(transactionPlan, witness, fullViewingKey, actionId).then(action => {
+    self.postMessage(action);
+  });
+};
+
+self.addEventListener('message', workerListener);
+
+async function executeWorker(
+  transactionPlan: TransactionPlan,
+  witness: WitnessData,
+  fullViewingKey: string,
+  actionId: number,
+): Promise<Action> {
```

Comment:
This is not JSON serialized. Think this type needs to be `JsonValue`.

Reply:
Also, in general, `JsonValue` does not communicate much. I think we should try to use `Jsonified<Action>` for the sake of code documentation.

Reply:
changed